app/src/storage/update-jenkins-server-state.ts

```diff
@@ -49,11 +51,28 @@ async function updateJenkinsServerState(
 			jenkinsServer
 		);
 	} catch (error) {
-		const errorMsg =
-			`Failed to update Jenkins server uuid ${uuid}`;
-		logger?.error(errorMsg);
+		logger?.error(`Failed to update Jenkins server uuid ${uuid}`);
 		throw error;
 	}
 }
 
-export { updateJenkinsServerState };
+async function updateJenkinsPluginConfigState(
+	uuid: string,
+	jenkinsEvent: JenkinsEvent,
+	logger: Logger
+): Promise<void> {
+	try {
+		const jenkinsServer = await storage.get(`${SERVER_STORAGE_KEY_PREFIX}${uuid}`);
+		const configProperties = ['ipAddress', 'autoBuildEnabled', 'autoBuildRegex', 'autoDeploymentsEnabled', 'autoDeploymentsRegex'];
+		jenkinsServer.pluginConfig = {
```

Comment:
this intentionally overrides plugingconfig instead off adding to it since we don't have a delete event and last in is truth at this point